rust-engine/src/backends/lean.rs

```diff
@@ -13,7 +19,16 @@ impl_doc_allocator_for!(LeanPrinter);
 
 impl Printer for LeanPrinter {
     fn resugaring_phases() -> Vec<Box<dyn Resugaring>> {
-        vec![]
+        vec![Box::new(BinOp::new(&[
```

Comment:
Why does this need to be boxed?

Reply:
`dyn Resugaring` is a dynamically sized type, so we need to box it!
`Vec` needs to know the size of the inner type.
Also, we can't put a reference here either: we may return resugaring that needs to allocate

Reply:
Ah, I didn't see the return type. Makes sense.